kernel/src/hil/framebuffer.rs

```diff
@@ -0,0 +1,143 @@
+//! Interface for FrameBuffer
+use crate::returncode::ReturnCode;
+use enum_primitive::cast::FromPrimitive;
+use enum_primitive::enum_from_primitive;
+
+enum_from_primitive! {
+    #[derive(Copy, Clone, PartialEq)]
+    pub enum ScreenRotation {
+        Normal = 0,
+        Rotated90 = 1,
+        Rotated180 = 2,
+        Rotated270 = 3,
+    }
+}
+
+impl From<ScreenRotation> for usize {
+    fn from(rotation: ScreenRotation) -> usize {
+        match rotation {
+            ScreenRotation::Normal => 0,
+            ScreenRotation::Rotated90 => 1,
+            ScreenRotation::Rotated180 => 2,
+            ScreenRotation::Rotated270 => 3,
+        }
+    }
+}
+
+enum_from_primitive! {
+    #[derive(Copy, Clone, PartialEq)]
+    #[allow(non_camel_case_types)]
+    pub enum ScreenColorDepth {
+        None = 0,
+        Mono = 1,
+        RGB_233 = 8,
```

Comment:
```suggestion
        /// Pixels encoded as 2-bit red channel, 3-bit green channel, 3-bit blue channel.
        RGB_233,
```

Something explaining what each of these modes are is probably appropriate for a HIL

Reply:
I create a new enum ``ScreenPixelFormat`` that allows the definition of various pixel formats. I added some, others can add other formats. I also created an impl for the enum with one function that returns the number of bits per fixel.